src/solver/conv_hip_implicit_gemm_grouped_fwd_xdlops.cpp

```diff
@@ -189,23 +193,143 @@ bool ConvHipImplicitGemmGroupFwdXdlops::CheckCKApplicability(
 }
 #endif
 
+std::vector<std::string> GetKernelAsTokens(const std::string& kernel)
+{
+    std::vector<std::string> tokens;
+    std::string token;
+    std::istringstream tokenStream(
+        kernel.substr(kernel.find('<') + 1, kernel.find('>') - kernel.find('<') - 1));
+    while(std::getline(tokenStream, token, ','))
+    {
+        token.erase(remove_if(token.begin(), token.end(), isspace),
+                    token.end()); // strip whitespace
+        tokens.push_back(token);
+    }
+    return tokens;
+}
+
+void PerformanceConfigHipImplicitGemmGroupFwdXdlops::InitHeuristicKernelIDs()
+{
+    for(int i = 0; i < valid_kernels.size(); i++)
+    {
+        if(valid_kernels[i].find("DeviceGroupedConvFwdMultipleD_Xdl_CShuffle") != std::string::npos)
+        {
+            heuristic_indexes.push_back(i);
+            heuristic_kernels.push_back(GetKernelAsTokens(valid_kernels[i]));
+        }
+    }
+}
+
+bool PerformanceConfigHipImplicitGemmGroupFwdXdlops::ModelApplyToken(int idx, std::string value)
+{
+
+    std::vector<int> new_heuristic_indexes;
+    new_heuristic_indexes.reserve(heuristic_indexes.size());
+    if(idx >= 5)
+        idx += 2; // skip MPerXDL and NPerXDL as they are constant
+    std::copy_if(
+        heuristic_indexes.begin(),
+        heuristic_indexes.end(),
+        std::back_inserter(new_heuristic_indexes),
+        [&](int heuristic_index) { return heuristic_kernels[heuristic_index][idx] == value; });
+    if(new_heuristic_indexes.empty())
+        return false;
+    heuristic_indexes = new_heuristic_indexes;
```

Comment:
It seems we don't need a temporary vector for `heuristic_indexes`.
Using `std::remove_if` and following `erase` if `remove_if` returned the last iterator is not equal to `begin()` or do nothing otherwise.